lib/descriptor/common.js

```diff
@@ -0,0 +1,259 @@
+ * and used for calculating checksum
+ * @const {String}
+ */
+
+common.INPUT_CHARSET = '0123456789()[],\'/*abcdefgh@:$%{}IJKLMNOPQRSTUVWXYZ' +
+                       '&+-.;<=>?!^_|~ijklmnopqrstuvwxyzABCDEFGH`#"\\ ';
+
+ /**
+ * Checksum character set for calculating checksum
+ */
+
+common.CHECKSUM_CHARSET = 'qpzry9x8gf2tvdw0s3jn54khce6mua7l';
+
+/**
+ * Test whether this descriptor is of scriptType
+ * ('pk', 'pkh', 'wpkh' etc.) or not
+ * @param {String} scriptType type of script
+ * @param {String} desc descriptor string
+ * @returns {Boolean}
+ */
+
+common.descType = function descType(scriptType, desc) {
+  if (
+    desc.length >= scriptType.length + 2 &&
+    desc[scriptType.length] === '(' &&
+    desc[desc.length - 1] === ')' &&
+    scriptType === desc.substring(0, scriptType.length)
+  ) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Strip outer level script string from descriptor
+ * @param {String} str
+ * @param {String} s
+ * @returns {String} stripped descriptor string
+ */
+
+common.strip = function strip(scriptType, desc) {
+  desc = desc.substring(scriptType.length + 1, desc.length - 1);
+  return desc;
+};
+
+/**
+ * Extract the expression that str begins with.
+ * This function will return the initial part of str, up to
+ * (but not including) the first comma or closing brace,
+ * skipping ones that are surrounded by braces. So for example,
+ * for "hello(world)(),(" the initial part
+ * "hello(world)()" will be returned. sp will be
+ * updated to skip the initial part that is returned.
+ * @param {String} str
+ * @returns {String}
+ */
+
+common.giveExpr = function giveExpr(str) {
+  let level = 0;
+  let it = 0;
+
+  while (it < str.length) {
+    if (str[it] === '(' || str[it] === '{') {
+      level++;
+    } else if (level && (str[it] === ')' || str[it] === '}')) {
+      level--;
+    } else if (
+      level === 0 &&
+      (str[it] === ')' || str[it] === '}' || str[it] === ',')
+    ) {
+      break;
+    }
+    it++;
+  }
+
+  const ret = str.slice(0, it);
+  return ret;
+};
+
+/**
+ * Test whether a string is a hex string or not
+ * @param {String} str
+ * @returns {Boolean}
+ */
+
+common.isHex = function isHex(str) {
+  const regexp = /^[0-9a-fA-F]+$/;
+  return regexp.test(str);
+};
+
+common.polyMod = function polyMod(c, val) {
+  const c0 = c.ushr(BN.from(35, 10, 'be'));
+  const c1 = BN.from('7ffffffff', 16, 'be');
+  c = c.and(c1).ushl(BN.from(5, 10, 'be')).xor(val);
+  if (c0.and(BN.from(1, 10, 'be')).toNumber())
+    c = c.xor(BN.from('f5dee51989', 16, 'be'));
+  if (c0.and(BN.from(2, 10, 'be')).toNumber())
+    c = c.xor(BN.from('a9fdca3312', 16, 'be'));
+  if (c0.and(BN.from(4, 10, 'be')).toNumber())
+    c = c.xor(BN.from('1bab10e32d', 16, 'be'));
+  if (c0.and(BN.from(8, 10, 'be')).toNumber())
+    c = c.xor(BN.from('3706b1677a', 16, 'be'));
+  if (c0.and(BN.from(16, 10, 'be')).toNumber())
+    c = c.xor(BN.from('644d626ffd', 16, 'be'));
+
+  return c;
+};
```

Comment:
I strongly feel that this should be an internal function and should not be exported because it will only ever be used in checksum creation and validation.
Same for constants `INPUT_CHARSET` and `CHECKSUM_CHARSET`